crates/worker/src/partition/snapshots/repository.rs

```diff
@@ -331,7 +331,7 @@ impl SnapshotRepository {
         level = "debug",
         skip_all,
         err,
-        fields(%partition_id),
+        fields(%partition_id, snapshot_id = tracing::field::Empty),
```

Comment:
Similar to my previous comment. I'd suggest you actually move those attributes to every log message to ensure we can see them in all levels and remove the fields from the span altogether.

Reply:
Hm, I did that all over, and really hated all the repetition everywhere. I also wanted to keep the `#[instrument(err)]` in place for `put` `get_latest` so I ended up keeping `partition_id` and `snapshot_id` as span-level attributes, and raised the span level to `Error` instead. But please let me know if you hate it and I'll move the attributes to all the events instead :-)